### DISCUSSION TITLE
Spawning Promises. Error handling issue.

### DISCUSSION THREAD

**Discussion by Sheldonfrith** [statelyai/xstate#1779] [Q&A] [answered]
I am aware the Docs recommend to "invoke" rather than "spawn" Promises, however if I "invoke" promises then, as far as I understand it, I cannot respond to their 'resolve'/'reject' if I exit the state they were "invoked" in. So I am spawning them as actors, which allows me to still handle them if they resolve after I have changed state. 

However I cannot figure out how to handle spawned promises that "reject". What is the equivalent to "onError" for actors? I've tried "error.invoke.<ID>" (which was just a guess) but that didn't work. All I can find via a google search is people with issues handling promise rejections for INVOKED promises, not SPAWNED promises. 

Any help would be appreciated. Thank you. I'm very excited about this library.

**Comment by davidkpiano** [comment]
Can you share a code reproduction of how you're currently spawning promises?

**Comment by Sheldonfrith** [comment]
For sure. Here is a basic example with just one promise actor: https://codesandbox.io/s/xstatespawnedpromises-8ipt2?file=/src/index.ts

**Comment by davidkpiano** [comment, marked as answer]
To handle errors for any spawned/invoked actor, you can use the `error()` function to create an `"error.platform.[name]"` string:

```js
import { error } from 'xstate/lib/actions';


// ...
[error("firstPromise")]: {
  actions: (context, event) => {
    console.log("promise error data:", event.data);
  }
},
```

This is likely to be improved in v5.

**Reply by Sheldonfrith** [reply]
Works perfectly.

Thank you for your fast replies.... its is really reducing the learning curve for me!

For anyone wondering where to put that code, here is the updated codesandbox with the errors handled....
https://codesandbox.io/s/xstatespawnedpromises-8ipt2?file=/src/index.ts
